src/DistributedLock.Azure/BlobClientWrapper.cs

```diff
@@ -25,9 +25,10 @@ public BlobClientWrapper(BlobBaseClient blobClient)
     {
         var conditions = new BlobRequestConditions { LeaseId = leaseId };
         var properties = SyncViaAsync.IsSynchronous
-            ? this._blobClient.GetProperties(conditions, cancellationToken)
-            : await this._blobClient.GetPropertiesAsync(conditions, cancellationToken).ConfigureAwait(false);
-        return properties.Value.Metadata;
+            ? _blobClient.GetProperties(conditions, cancellationToken).AsTask()
+            : _blobClient.GetPropertiesAsync(conditions, cancellationToken);
+        return (await properties.ConfigureAwait(false)).Value.Metadata;
```

Comment:
I'm not sure I see the optimization. In the sync path, we've added an allocation (for `AsTask()`) where before we didn't have one. In the async path we have the same number of `await`s as before. Can you explain further? Maybe provide some benchmark results that show the improvement?